src/clib/pio_darray.c

```diff
@@ -482,21 +493,20 @@ int PIOc_write_darray(const int ncid, const int vid, const int ioid,
  * @param array: pointer to the data to be read. This is a
  * pointer to the distributed portion of the array that is on this
  * processor.
- *
  * @return 0 for success, error code otherwise.
  * @ingroup PIO_read_darray
  */
 int PIOc_read_darray(const int ncid, const int vid, const int ioid,
                      const PIO_Offset arraylen, void *array)
 {
     iosystem_desc_t *ios;  /* Pointer to io system information. */
-    file_desc_t *file;
-    io_desc_t *iodesc;
-    void *iobuf = NULL;
-    size_t rlen = 0;
-    int tsize; /* Total size. */
+    file_desc_t *file;     /* Pointer to file information. */
+    io_desc_t *iodesc;     /* Pointer to IO description information. */
+    void *iobuf = NULL;    /* ??? */
```

Comment:
iobuf holds the data as read on the io node. 
rlen is the length of that buffer.